spec/betagouvbot/mailer_spec.rb

```diff
@@ -2,65 +2,31 @@
 # frozen_string_literal: true
 
 RSpec.describe BetaGouvBot::Mailer do
-  let(:in_3w) do
-    instance_double(
-      'mail',
-      format: { personalizations: [to: ['email' => 'ann@email.coop']] }
-    )
-  end
-
-  let(:in_2w) do
-    instance_double(
-      'mail',
-      format: {
-        personalizations: [
-          to: [{ 'email' => 'ann@email.coop' }, { 'email' => 'hi@email.coop' }]
-        ]
-      }
-    )
-  end
-
-  let(:rules) do
-    {
-      1  => { mail: instance_double('mail') },
-      14 => { mail: in_2w },
-      21 => { mail: in_3w }
-    }
-  end
+  let(:horizons) { BetaGouvBot::NotificationRule.horizons }
+  let(:schedule) { BetaGouvBot::Anticipator.(authors, horizons, Date.today) }
 
-  let(:schedule) { BetaGouvBot::Anticipator.(authors, rules.keys, Date.today) }
-
-  before { allow(described_class).to receive(:post) }
+  before { allow(described_class).to receive(:email) }
 
   describe 'selecting recipients of emails' do
-    context 'when a member has an end date in three weeks' do
-      let(:authors) { [id: 'ann', fullname: 'Ann', end: (Date.today + 21).iso8601] }
-
-      it 'sends an email directly to the author' do
-        described_class.(schedule, rules).map(&:execute)
-        expect(described_class).to have_received(:post)
-          .with(request_body: in_3w.format)
-      end
-    end
+    before { allow(described_class).to receive(:rule) { rule } }
 
-    context 'when a member has an end date in two weeks' do
-      let(:authors) { [id: 'ann', fullname: 'Ann', end: (Date.today + 14).iso8601] }
+    let(:authors) { [id: 'ann', fullname: 'Ann', end: (Date.today + 21).iso8601] }
+    let(:rule)    { BetaGouvBot::NotificationRule.find(horizon: 21) }
 
-      it 'sends an email to the author and contact' do
-        described_class.(schedule, rules).map(&:execute)
-        expect(described_class).to have_received(:post)
-          .with(request_body: in_2w.format)
-      end
+    it 'sends an email directly to the author' do
+      described_class.(schedule)
+      expect(described_class).to have_received(:email)
```

Comment:
Now that Mailer returns a value (an array of MailAction), it seems to me that testing for the returned value is preferable over spying. It's also more consistent with what the other specs are doing. If you think there's a strategic reason to prefer spying over testing returned values against expected values, I'd be keen to understand it and help switch over the other specs to that standard.